File: problem22.rb

string_of_names = File.open("/home/rgarcia5/Desktop/project_euler/p022_names.txt")

list_of_names = string_of_names.read.split(",").sort
alphabet = ("A".."Z").to_a


result = 0
list_of_names.each do |name|
  
  name.each_char do |letter|
    next unless alphabet.include?(letter)
      result += (alphabet.index(letter) + 1) * (list_of_names.index(name) + 1)
    end

end

p result

File: problem22.py
import string
def get_names():
  with open("p022_names.txt", mode = "r") as file:
    result = file.read().split(",")
  file.closed
  return sorted(map(lambda x: x.replace("\"", ""), result))


alphabet = list(string.ascii_uppercase)

def names_scores():
  list_of_names = get_names()
  score = 0
  for name in list_of_names:
    for letter in name:
      score += (alphabet.index(letter) + 1) * (list_of_names.index(name) + 1)
  return score

print names_scores()
